src/scripts/eastern_kingdoms/stranglethorn_vale/zulgurub/boss_mandokir.cpp

```diff
@@ -261,6 +252,31 @@ struct boss_mandokirAI : public ScriptedAI
         }
     }
 
+    void CheckVilebranchState(bool reset = false)
+    {
+        // If Vilebranch dies and group wipes, boss should start at the bottom of the stairs
+        // Video: https://www.youtube.com/watch?v=joaWY0wjOXI
+        Creature* vileBranch = m_creature->FindNearestCreature(11391, 100.0f, true);
+        bool isVilebranchDead = !vileBranch || !vileBranch->isAlive();
+        if (reset || m_VilebranchDead != isVilebranchDead)
+        {
+            if (isVilebranchDead)
+            {
+                m_creature->RemoveFlag(UNIT_FIELD_FLAGS, START_FLAGS);
+                m_creature->SetHomePosition(-12195.0f, -1948.0f, 130.0f, 3.14f);
+                m_creature->GetMotionMaster()->MoveTargetedHome();
+            }
+            else
+            {
+                m_creature->SetFlag(UNIT_FIELD_FLAGS, START_FLAGS);
+                m_creature->ResetHomePosition();
+            }
+            m_creature->GetMotionMaster()->MoveTargetedHome();
```

Comment:
You call this twice in one of the code paths

Reply:
oops